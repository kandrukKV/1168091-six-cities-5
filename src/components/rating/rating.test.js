import React from "react";
import renderer from "react-test-renderer";
import Rating from "./rating";

const className = `rating-class`;

describe(`Rating render correctly`, () => {
  it(`Rating is 0`, () => {
    const tree = renderer.create(
        <Rating
          className={className}
          rating={0}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Rating is 1`, () => {
    const tree = renderer.create(
        <Rating
          className={className}
          rating={1}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Rating is 2`, () => {
    const tree = renderer.create(
        <Rating
          className={className}
          rating={2}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Rating is 3`, () => {
    const tree = renderer.create(
        <Rating
          className={className}
          rating={3}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Rating is 4`, () => {
    const tree = renderer.create(
        <Rating
          className={className}
          rating={4}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Rating is 5`, () => {
    const tree = renderer.create(
        <Rating
          className={className}
          rating={5}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
