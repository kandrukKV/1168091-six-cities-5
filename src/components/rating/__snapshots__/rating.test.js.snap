// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating render correctly Rating is 0 1`] = `
<div
  className="rating-class__rating rating"
>
  <div
    className="rating-class__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "0%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;

exports[`Rating render correctly Rating is 1 1`] = `
<div
  className="rating-class__rating rating"
>
  <div
    className="rating-class__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "20%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;

exports[`Rating render correctly Rating is 2 1`] = `
<div
  className="rating-class__rating rating"
>
  <div
    className="rating-class__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;

exports[`Rating render correctly Rating is 3 1`] = `
<div
  className="rating-class__rating rating"
>
  <div
    className="rating-class__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "60%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;

exports[`Rating render correctly Rating is 4 1`] = `
<div
  className="rating-class__rating rating"
>
  <div
    className="rating-class__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "80%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;

exports[`Rating render correctly Rating is 5 1`] = `
<div
  className="rating-class__rating rating"
>
  <div
    className="rating-class__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;
